mapstr/mapstr_test.go

```diff
@@ -977,8 +977,13 @@ func TestMapStrJSONLog(t *testing.T) {
 	}
 	expectedJSON := string(data)
 
-	logp.NewLogger("test").Infow("msg", "m", m)
-	logs := logp.ObserverLogs().TakeAll()
+	observedCore, observedLogs := observer.New(zapcore.DebugLevel)
+	logger := logptest.NewTestingLogger(t, "test", zap.WrapCore(func(core zapcore.Core) zapcore.Core {
+		return observedCore
+	}))
```

Comment:
I'm wondering if it'd be helpful to have a function on logptest that returns this logger. What do you think?
I believe it'd convenient and help us to converge to a single way of using it in tests.